src/caja-application.c

```diff
@@ -267,26 +267,90 @@ open_window (CajaApplication *application,
+{
+    CajaApplication *self = CAJA_APPLICATION (application);
+    CajaWindow *window;
+    gchar *uri = NULL;
+
+    /*monitor the preference to use browser or spatial windows */
+    /*connect before trying to read or this preference won't be read by root or after change*/
+     g_signal_connect_swapped(caja_preferences, "changed::"CAJA_PREFERENCES_ALWAYS_USE_BROWSER,
+                      G_CALLBACK (caja_empty_callback_to_ensure_read),
+                      self);
+
+    if (browser_window ||g_settings_get_boolean (caja_preferences, CAJA_PREFERENCES_ALWAYS_USE_BROWSER)) {
+        window = caja_application_create_navigation_window (application,
+                 screen);
+    } else {
+        window = caja_application_get_spatial_window (application,
+                 NULL,
+                 NULL,
+                 locations[0],
+                 screen,
+                 NULL);
+    }
+
+    /* open all locations */
+    uri = g_file_get_uri (locations[0]);
+    g_debug("Opening new tab at uri %s\n", uri);
+    caja_window_go_to (window, locations[0]);
+    for (int i = 1; i< n_files;i++){
+        /* open tabs in reverse order because each 
+         * tab is opened before the previous one */
+        guint tab = n_files-i;
+        uri = g_file_get_uri (locations[tab]);
+        g_debug("Opening new tab at uri %s\n", uri);
+        if(i==0){
+            caja_window_go_to (window, locations[tab]);
+        }else{
+            caja_window_go_to_tab (window, locations[tab]);
+        } 
+    }
+
+    if (geometry != NULL && !gtk_widget_get_visible (GTK_WIDGET (window))) {
+        /* never maximize windows opened from shell if a
+         * custom geometry has been requested.
+         */
+        gtk_window_unmaximize (GTK_WINDOW (window));
+        eel_gtk_window_set_initial_geometry_from_string (GTK_WINDOW (window),
+                                 geometry,
+                                 APPLICATION_WINDOW_MIN_WIDTH,
+                                 APPLICATION_WINDOW_MIN_HEIGHT,
+                                 FALSE);
+    }
+
+    g_free (uri);
+}
+
 static void
 open_windows (CajaApplication *application,
           GFile **files,
           GdkScreen *screen,
           const char *geometry,
-          guint len,
-          gboolean browser_window)
+          guint n_files,
+          gboolean browser_window,
+          gboolean open_in_tabs)
 {
     guint i;
 
     if (files == NULL || files[0] == NULL) {
         /* Open a window pointing at the default location. */
-        open_window (application, NULL, screen, geometry, browser_window );
+        open_window (application, NULL, screen, geometry, browser_window);
     } else {
-        /* Open windows at each requested location. */ 
-        i = 0;
-        while (i < len ){ 
-              open_window (application, files[i], screen, geometry, browser_window);
-               i++ ;
-         }
+        if(open_in_tabs){
```

Comment:
same as before

Reply:
here it's the old code which was wrong idnented

Reply:
I meant this new line `if(open_in_tabs){`  --> `if (open_in_tabs) {`